crates/cli/src/benchmark.rs

```diff
@@ -166,25 +185,48 @@ impl InProcessBenchmarkCommand {
         let bytes = fs::read(&self.wasmfile).context("Attempting to read Wasm bytes")?;
         log::debug!("Wasm benchmark size: {} bytes", bytes.len());
 
-        let mut measurements = Vec::with_capacity(self.iterations);
+        let arch = this_arch();
+        let engine = self.engine.to_string();
+        let wasm = self.wasmfile.display().to_string();
+        let mut measurements = Measurements::new(arch, &engine, &wasm);
+
+        let mut measure = self.measure.build();
+
         for _ in 0..self.iterations {
-            measurements.push(benchmark(&mut bench_api, &bytes, self.measure)?);
+            benchmark(&mut bench_api, &bytes, &mut measure, &mut measurements)?;
+            measurements.next_iteration();
         }
 
+        let measurements = measurements.finish();
+
         match self.output_format {
             OutputFormat::Json => {
                 println!("{}", serde_json::to_string(&measurements)?);
```

Comment:
If the "flatten to CSV rows" logic is separated from the recording logic, then we have the option of reporting the data hierarchically with JSON and flattened with CSV. I don't know if we *should* do that, but one thing that comes to mind is that the option to report things hierarchically will make it significantly easier to interface with the current history web service (which accepts hierarchical data in a slightly different format)--less to rewrite.